shm/src/Cargo.cpp

```diff
@@ -0,0 +1,27 @@
+#include "shm/inc/Cargo.hpp"
+
+#include <iostream>
+
+Cargo::Cargo(const std::string& name, size_t amount, size_t basePrice) 
+    : name_(name)
+    , amount_(amount)
+    , basePrice_(basePrice)
+{}
+
+Cargo& Cargo::operator+=(const size_t amount) {
+    amount_ += amount;
+    return *this;
+}
+
+Cargo& Cargo::operator-=(const size_t amount) {
+    if (amount <= amount_) {
+        amount_ -= amount;
+    } else {
+        std::cerr << "Amount of cargo can't be under 0\n";
```

Comment:
You only inform about problem, but you don't handle error case. Maybe throw exception, or set amount_ as std::numeric_limits::max, to inform user of this operator that sth really bad happend. Notice that game should continiue, so you need to handle somehow that situations :)